src/new/helenos/fibril_synch.rs

```diff
@@ -0,0 +1,49 @@
+//! HelenOS fibril synchronization primitives
+//!
+//! * Header file: <https://github.com/HelenOS/helenos/tree/master/uspace/lib/c/include/fibril_synch.h>
+
+pub use crate::fibril::*;
+use crate::{
+    c_int,
+    errno_t,
+    list_initialize,
+    list_t,
+};
+
+s! {
+    pub struct fibril_mutex_t {
+        pub oi: fibril_owner_info_t,
+        pub counter: c_int,
+        pub waiters: list_t,
+    }
+
+    pub struct fibril_condvar_t {
+        pub waiters: list_t,
+    }
+}
+
+f! {
+    pub fn fibril_mutex_initialize(fm: *mut fibril_mutex_t) -> () {
+        let fm = &mut *fm;
+        fm.oi.owned_by = core::ptr::null_mut();
+        fm.counter = 1;
+        list_initialize(&mut fm.waiters);
+    }
```

Comment:
https://github.com/HelenOS/helenos/blob/144fafda7cb6d43dfda1c93b8e46b8d51635e0c2/uspace/lib/c/include/fibril_synch.h#L55-L66 seems to effectively return a definition. Any reason not to do similar here and directly return a `fibril_mutex_t`, rather than using a pointer?

You may want to uppercase to match the definitions too.